src/main-dev.rs

```diff
@@ -1,8 +1,3 @@
-// I don't see **any** reason for this waring to enabled. We only build binary
-// and most of these warnings do not apply to real visibility of types inside
-// the crate
```

Comment:
The logic is that private types cannot be leaked from modules they are defined in, so if you see a private type you have a guarantee that it's restricted to this module.
These crate wants to leak private types like `UpstreamValidator` from their module, therefore the error.

If `UpstreamValidator` and friends are intended to be implementation details (**1**), but still leaked in some form (**2**) (e.g. through aliases), they have to be marked as `pub` (because **2**), but made "unnameable" by placing them into a private module (because **1**).
(The [tracking issue](https://github.com/rust-lang/rust/issues/34537) for `private_in_public` contains a similar example.)